Gems/Atom/Feature/Common/Code/Source/SplashScreen/SplashScreenPass.cpp

```diff
@@ -0,0 +1,147 @@
+
+        if (!(settingsRegistry && settingsRegistry->GetObject(&m_settings, azrtti_typeid(m_settings), setregPath.c_str())))
+        {
+            return;
+        }
+
+        m_splashScreenImage = AZ::RPI::LoadStreamingTexture(m_settings.m_imagePath);
+        m_screenTime = m_settings.m_lastingTime;
+
+        if (!m_splashScreenImage)
+        {
+            return;
+        }
+
+        m_splashScreenImageIndex.Reset();
+        m_splashScreenImageIndex.Reset();
+
+        const TimeUs currentTimeUs = static_cast<TimeUs>(AZStd::GetTimeNowMicroSecond());
+        m_lastRealTimeStamp = aznumeric_cast<float>(currentTimeUs) / 1000000.0f;
+
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void SplashScreenPass::Clear()
+    {
+        m_splashScreenImage = nullptr;
+
+        m_splashScreenImageIndex.Reset();
+        m_splashScreenImageIndex.Reset();
+    }
+
+    void SplashScreenPass::FrameBeginInternal([[maybe_unused]] FramePrepareParams params)
+    {
+        FullscreenTrianglePass::FrameBeginInternal(params);
+
+        if (m_beginTimer)
+        {
+            // This is not ideal to stop other systems. It may cause conflicts when multiple sources are trying to set the value.
+            if (auto* timeSystem = AZ::Interface<ITime>::Get())
+            {
+                timeSystem->SetSimulationTickScale(0.0f);
+            }
+        }
+    }
+
+    void SplashScreenPass::FrameEndInternal()
+    {
+        FullscreenTrianglePass::FrameEndInternal();
+
+        if (m_screenTime < 0.0f)
+        {
+            m_beginTimer = false;
+
+            // Disable the pass after life time passes.
+            SetEnabled(false);
+
+            // Reset the tick scale
+            if (auto* timeSystem = AZ::Interface<ITime>::Get())
+            {
+                timeSystem->SetSimulationTickScale(1.0f);
+            }
+        }
+    }
+
+    void SplashScreenPass::CompileResources(const AZ::RHI::FrameGraphCompileContext& context)
+    {
+        m_shaderResourceGroup->SetImage(m_splashScreenImageIndex, m_splashScreenImage);
+        m_shaderResourceGroup->SetConstant(m_splashScreenParamsIndex, m_splashScreenParams);
+
+        AZ::RPI::FullscreenTrianglePass::CompileResources(context);
+    }
+
+    void SplashScreenPass::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        // Not using the delta time from Tick bus because it could be scaled.
+        const TimeUs currentTimeUs = static_cast<TimeUs>(AZStd::GetTimeNowMicroSecond());
+        float currentRealTimeStamp = aznumeric_cast<float>(currentTimeUs) / 1000000.0f;
+        float realDeltaTime = currentRealTimeStamp - m_lastRealTimeStamp;
+        m_lastRealTimeStamp = currentRealTimeStamp;
+
+        if (m_beginTimer)
+        {
+            m_screenTime -= realDeltaTime;
+        }
+
+        if (m_settings.m_fading)
+        {
+            m_splashScreenParams.m_fadingFactor = m_screenTime < 0.0f ? 0.0f : m_screenTime / 10.0f;
```

Comment:
It looks like the duration is hard-coded to 10 seconds here
```suggestion
            m_splashScreenParams.m_fadingFactor = m_screenTime < 0.0f ? 0.0f : m_screenTime / m_settings.m_lastingTime;
```
maybe a `1-x^3` curve would be nice?
```suggestion
            m_splashScreenParams.m_fadingFactor = m_screenTime < 0.0f ? 0.0f : (1.f - powf(m_screenTime / m_settings.m_lastingTime, 3.f));
```

Reply:
I was originally trying to fade out the image.
It actually doesn't matter that much. I mainly used the fading factor as a case of how to manipulate the shader behavior for this pass. But lacking of a use case made it hard to imagine what we need. For example, we can have multiple images display in a row, accepting input to skip the splash screen. Or we may need animation for scrolling text.